ert3/stats/_stats.py

```diff
@@ -43,14 +43,11 @@ def index(self) -> ert.data.RecordIndex:
         return self._index
 
     def _to_record(self, x: np.ndarray) -> ert.data.Record:  # type: ignore
```

Comment:
,this ignore exists due to `.mypy-strict.ini` where generic types need to be more specific. The type of `x` should be `numpy.typing.NDArray[typing.Union[numpy.float64, numpy.int_]]`, but this cannot be implemented, since `numpy.typing` module is introduced in version `1.20`. So the question is are we willing to upgrade `numpy` just for  typing.

Reply:
this counts for all the `ignore`s in this file except for line

Reply:
`numpy` is defined without an upper bound on version in our `setup.py`, which means `ert` will try to use the latest version.
What do you mean by upgrading `numpy`?

Reply:
don't know why, but the latest version of python `3.6` is numpy `1.19`. and even if you try to `pip install -U numpy` the response is that numpy is up to the latest version. I know that our workflow for typing uses python `3.8` so I could use a python version guard to create the type only for `typing_extensions.TYPE_CHECKING` and `python version greater than python `3.6`, but I wouldn't that be overkill for typing?

Reply:
furthermore, the above suggestion wouldn't be viable
[Variables vs type aliases](https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases)
>Type aliases can’t be defined conditionally (unless using supported Python version and platform checks):

Reply:
> `numpy` is defined without an upper bound on version in our `setup.py`, which means `ert` will try to use the latest version. What do you mean by upgrading `numpy`?

```bash
❯ python -V
Python 3.6.9
❯ pip install numpy==1.20
ERROR: Could not find a version that satisfies the requirement numpy==1.20 (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0rc1, 1.13.0rc2, 1.13.0, 1.13.1, 1.13.3, 1.14.0rc1, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0rc1, 1.15.0rc2, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0rc1, 1.16.0rc2, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0rc1, 1.17.0rc2, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0rc1, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0rc1, 1.19.0rc2, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5)
ERROR: No matching distribution found for numpy==1.20
```